### PR TITLE
feat: Report as issue markdown

### LINKED ISSUE #11: Add a warning if we are removing all users from CODEOWNERS

### PR DESCRIPTION
**Pull request by zkoppert**
# Pull Request

fixes #11 

## Proposed Changes

This pull request introduces a new feature that generates a report of users removed from the `CODEOWNERS` files across the repos that were affected. The report is written in Markdown format and can be optionally created as an issue in the repository. The `ISSUE_REPORT` environment variable controls this feature. The `README.md` file has been updated to explain this new feature and how to use it. 

I also added/modified tests to cover the new functionality.

### Author/Contributor

- [x] If documentation is needed for this change, has that been included in this pull request
- [x] run `make lint` and fix any issues that you have introduced
- [x] run `make test` and ensure you have test coverage for the lines you are introducing
- [x] If publishing new data to the public (scorecards, security scan results, code quality results, live dashboards, etc.), please request review from `@jeffrey-luszcz`

### Reviewer

- [x] Label as either `fix`, `documentation`, `enhancement`, `infrastructure`, `maintenance` or `breaking`

### CODE REVIEW

File: cleanowners.py [hunk 1]
```diff
@@ -182,7 +205,7 @@ def get_repos_iterator(organization, repository_list, github_connection):
     return repos
 
 
-def get_usernames_from_codeowners(codeowners_file_contents):
+def get_usernames_from_codeowners(codeowners_file_contents, ignore_teams=True):
     """Extract the usernames from the CODEOWNERS file"""
     usernames = []
     for line in codeowners_file_contents.decoded.splitlines():
```
Comment (zkoppert) [inline cleanowners.py:211]: I will open up a new PR to turn this into a more efficient regex instead of line by line.

File: env.py [hunk 2]
```diff
@@ -155,6 +157,15 @@ def get_env_vars(
             "Remove users no longer in this organization from CODEOWNERS file"
         )
 
+    issue_report = os.getenv("ISSUE_REPORT")
```
Comment (jmeridth) [inline env.py:160]: Do we want to reuse [this helper function](https://github.com/github/evergreen/blob/b52dc5889ba82ea6bae2e30b11b3570921f01994/env.py#L12-L25) in evergreen here or in another PR?

**Review by jmeridth — APPROVED** [review]
LGTM, just one question.  Code is very readable.

File: env.py (hunk 2 above)
Comment (zkoppert) [reply env.py:160]: Lets do that here. I think its related enough since we are adding duplicate code.

Comment (jmeridth) [reply env.py:160]: On it.  I'll push commit

Comment (jmeridth) [reply env.py]: This applied to `ISSUE_REPORT` and `DRY_RUN` env vars

**Comment by jmeridth** [comment]
Ran lint locally and got `cleanowners.py:145:58: E0601: Using variable 'codeowners_file_contents_new' before assignment (used-before-assignment)` hence the change to default `codeowners_file_contents_new` to `None` in my commit.

**Comment by zkoppert** [comment]
Looks good! Thanks for adding @jmeridth. Leaving this pr open for @sutterj to take a look at.

**Review by sutterj — APPROVED** [review]

**Pull request merged**